service/history/queueProcessor.go

```diff
@@ -210,7 +212,15 @@ processorPumpLoop:
 			// use a separate gorouting since the caller hold the shutdownWG
 			go p.Stop()
 		case <-p.notifyCh:
-			p.processBatch()
+			if !p.isPriorityTaskProcessorEnabled() || p.redispatchQueue.Len() <= p.options.MaxRedispatchQueueSize() {
```

Comment:
What is isPriorityTaskProcessorEnabled for?

Reply:
This is the feature flag for controlling whether the new task processor based on priority will be used or not. If it's not enabled, then basically there's no re-dispatch queue and task loading won't be skipped.